controllers/controllers.go

```diff
@@ -126,10 +149,15 @@ func (r *instanceReconciler) upgradeBlocked(node *core.Node, upgradingToCSI bool
 	if value := node.GetLabels()[metadata.CSIConfiguredLabel]; value == "true" {
 		return false
 	}
-	if len(node.Status.VolumesAttached) == 0 {
-		return false
+	// For each pod running on this node, check if there is a PVC backed volume, if one is found, block the upgrade
+	for _, pod := range nodeSummary.Pods {
```

Comment:
If you care only about `nodeSummary.Pods` then there is no need to pass the whole object.